packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -49,6 +49,7 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
 
     autoSave: 'on' | 'off' = 'on';
     autoSaveDelay: number = 500;
+    suppressOpenEditorWhenDirty = false;
```

Comment:
Can you clarify what `suppressOpenEditorWhenDirty` refers to and why it is necessary when we already have 
`this.suppressedOpenIfDirty`?

Reply:
This solution is what I referred to in https://github.com/eclipse-theia/theia/pull/8910#discussion_r576675969
If model itself contains indication to suppress opening of editor - then I do not need to check using ApplicationShell (!CustomEditorWidget.is(this.shell.activeWidget)). Other than this way, I'm not sure how to do it otherwise.